src/main/java/ui/components/FilterTextField.java

```diff
@@ -32,6 +36,10 @@
     public static final String VALID_FILTER_STYLE = "-fx-control-inner-background: white";
     public static final String INVALID_FILTER_STYLE = "-fx-control-inner-background: #EE8993";
 
+    // For on-the-fly parsing and checking
+    private final ValidationSupport validationSupport = new ValidationSupport();
+    private final SuggestionMenu suggestion;
```

Comment:
Perhaps we should use `Suggestions` everywhere when we refer to something containing multiple suggestions?